src/Neo/SmartContract/Native/NamedCurveHash.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// NamedCurveHash.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+namespace Neo.SmartContract.Native
+{
+    /// <summary>
+    /// Represents a pair of the named curve used in ECDSA and a hash algorithm used to hash message.
+    /// This is a compatible extension of an obsolete <see cref="NamedCurve"/> enum.
+    /// </summary>
+    public enum NamedCurveHash : byte
+    {
+        /// <summary>
+        /// The secp256k1 curve and SHA256 hash algorithm.
+        /// </summary>
+        secp256k1SHA256 = 22,
```

Comment:
Why don't reuse the same enum, just rename them?

Reply:
per @Jim8y request.

Reply:
It was implemented like that before. But then I’ve changed the code due to https://github.com/neo-project/neo/pull/3209#discussion_r1590570107.

In general, I’m OK with a simple renaming, because I doubt that there are other users of this enum except the devpack, and devpack has its own enum. But probably Jimmy is aware of such users.